scripts/dev/e2e.py

```diff
@@ -78,17 +78,19 @@ def _prepare_testrunner_environment(config_file: str):
     )
 
 
-def create_kube_config():
+def create_kube_config() -> None:
     """Replicates the local kubeconfig file (pointed at by KUBECONFIG),
     as a ConfigMap."""
     corev1 = client.CoreV1Api()
     print("Creating kube-config ConfigMap")
 
     svc = corev1.read_namespaced_service("kubernetes", "default")
-    kube_config = os.getenv("KUBECONFIG")
-    with open(kube_config) as fd:
+    kube_config_path = os.getenv("KUBECONFIG")
+    assert kube_config_path is not None
```

Comment:
rather than an assertion, I think it would be better to explicitly throw an exception (not `AssertionError`). We should reserve assert statements for test code. In this case there was an error setting up our environment, rather than a test failure.

maybe

```python
if kube_config_path is None:
   raise ValueError("kube_config_path must not be None")
```